code/weapon/weapons.cpp

```diff
@@ -4109,17 +4134,27 @@ void find_homing_object_cmeasures_1(object *weapon_objp)
 						// heat seeker and javelin HS this likely to chase a countermeasure
 						chance = cm_wip->cm_heat_effectiveness/wip->seeker_strength;
 					}
-					if ((objp->signature != wp->cmeasure_ignore_objnum) && (objp->signature != wp->cmeasure_chase_objnum))
-					{
-						if (frand() >= chance) {
-							wp->cmeasure_ignore_objnum = objp->signature;	//	Don't process this countermeasure again.
-						} else  {
-							wp->cmeasure_chase_objnum = objp->signature;	//	Don't process this countermeasure again.
+
+					if (The_mission.ai_profile->flags2 & AIPF2_COUNTERMEASURES_DECOY_ONCE) {
+						wp->cmeasure_ignore_list->push_back(objp->signature);
+					}
+					else {
+						// emulate retail by only "remembering" one seen CM
+						if (wp->cmeasure_ignore_list->size() == 0) {
```

Comment:
Yes, I'll fix pending an answer below about removing the AI profiles flag entirely.